atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -165,6 +166,34 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
     public TransactionAndImmutableTsLock setupRunTaskWithConditionThrowOnConflict(PreCommitCondition condition) {
         StartIdentifiedAtlasDbTransactionResponse transactionResponse
                 = timelockService.startIdentifiedAtlasDbTransaction();
+        return wrap(condition, transactionResponse);
+    }
+
+    @Override
+    public List<TransactionAndImmutableTsLock> setupRunTaskWithConditionThrowOnConflictBatch(
```

Comment:
I would like to know if we could make TransactionAndImmutableTsLock have a #close method that correctly invokes the #finishMethod